cli/pgx: Update pgx versions in go.mod

spilchen opened the pull request:
This commit updates the versions of specific packages from the github.com/jackc repository in go.mod to address known security vulnerabilities. The updated versions include fixes for these vulnerabilities.

Notably, the signature of the Encode() function in the pgx library has changed. It now returns both the encoded message and an error. Most of the changes in this commit are updates to handle the new function signature appropriately.

Epic: none
Release note: none

Informs #137595

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139715)
<!-- Reviewable:end -->

spilchen committed:
cli/pgx: Update pgx versions in go.mod

This commit updates the versions of specific packages from the
github.com/jackc repository in go.mod to address known security
vulnerabilities. The updated versions include fixes for these
vulnerabilities.

Notably, the signature of the Encode() function in the pgx library has
changed. It now returns both the encoded message and an error. Most of
the changes in this commit are updates to handle the new function
signature appropriately.

Epic: none
Release note: none

Informs #137595

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 18 of 18 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139715#-:-OHO_0p33DYDmbdfrbs5:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
